parser_library/src/processing/opencode_provider.cpp

```diff
@@ -419,11 +446,13 @@ void opencode_provider::convert_ainsert_buffer_to_copybook()
     auto virtual_copy_name =
         m_ctx->hlasm_ctx->ids().add("AINSERT:" + std::to_string(m_ctx->hlasm_ctx->obtain_ainsert_id()));
 
-    auto new_file = m_virtual_files.try_emplace(virtual_copy_name, std::move(result)).first;
+    auto new_file = m_virtual_files.try_emplace(virtual_copy_name, std::move(result), next_virtual_file_id()).first;
```

Comment:
It feels a little bit weird to deal with the virtual file IDs. Consider moving this responsibility somewhere else.

Reply:
I've actually realized that there is a much bigger (lifetime) issue, so I will rework this.